### PR TITLE
use custom emulator

### LINKED ISSUE #520: Flow emulator cannot be started with custom emulator
### Problem

When emulator name is changed from "default" to something else, we get an error message saying “no default emulator account”. This means that currently flow cli users can’t start emulator with their custom emulator. 

### Steps to Reproduce

- change the emulator name from “default” to something else 
- start emulator

### Acceptance Criteria

Users should be able to start emulator with non default emulator.

### PR DESCRIPTION
**Pull request by sukantoraymond**
Closes #520

This PR enables user to start emulator with non default emulator in flow.json

______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by sukantoraymond** [comment]
As I am working on this PR, I'm just thinking more about the reasons for a user to create a custom emulator in flow.json. If it's to use a different port number, current implementation of using different port number in flow.json won't work since currently it is using the default value in start.go in emulator repo. We can just pass the custom port number using flag instead of using flow.json. 

If the user wants to change the emulator service account, this means that the user will have to create a new emulator account. Currently from emulator repo code, seems that the address used in emulator is default, so changes in flow.json won't even be reflected unless we make changes to emulator repo code to use the custom address value in flow.json for emulator. 

My concluding thought I guess is if we even want to enable users to use custom emulator? @sideninja  

Screen Shot 2022-08-23 at 6 30 46 PM

**Comment by sukantoraymond** [comment]
What this PR does is basically just change the name of the default emulator. If we want to use custom values like port, address on emulator, we would need to make changes on the emulator repo.

File: pkg/flowkit/state.go [hunk 1]
```diff
@@ -173,11 +173,13 @@ func (p *State) SetAccountFileLocation(account Account, location string) {
 
 // EmulatorServiceAccount returns the service account for the default emulator profile.
 func (p *State) EmulatorServiceAccount() (*Account, error) {
-	emulator := p.conf.Emulators.Default()
-	if emulator == nil {
-		return nil, fmt.Errorf("no default emulator account")
-	}
+	emulator := config.Emulator{}
```
Comment (sideninja) [inline pkg/flowkit/state.go:176]:
```suggestion
	 var emulator config.Emulator
```

File: pkg/flowkit/state.go [hunk 2]
```diff
@@ -173,11 +173,13 @@ func (p *State) SetAccountFileLocation(account Account, location string) {
...
 
+	if len(p.conf.Emulators) > 0 && p.conf.Emulators[0].Name != config.DefaultEmulatorConfigName {
```
Comment (sideninja) [inline pkg/flowkit/state.go:178]: Why do we need to change this? Not sure how would the first emulator be important in case there are multiple defined?

**Comment by sideninja** [comment]
We could set the private key using the configuration, but I agree. I think we should remove tool-specific configurations from flow.json.

**Comment by sukantoraymond** [comment]
I think we can just add a message saying "default emulator is required" is we detect a custom emulator in flow.json

**Comment by sideninja** [comment]
I think we can close both for now.

**Pull request closed without merging**